### PR TITLE
KokkosBatched - add specialization for float2, float4 and double4.

### PR DESCRIPTION
**Pull request by kyungjoo-kim**
These are for testing two scalar type approach in ifpack2 running spotcheck on white.

### CODE REVIEW

**Comment by kyungjoo-kim** [comment]
@ndellingwood This is the PR that I mentioned just before. The conflicting files are a result from git mv. I changed the location.

**Comment by kyungjoo-kim** [comment]
```
[kyukim@white26 spotcheck]$ . ../spotcheck.sh
Running on machine: white
Going to test compilers:  cuda/9.2.88
Testing compiler cuda/9.2.88
  Starting job cuda-9.2.88-Cuda_OpenMP-release
  PASSED cuda-9.2.88-Cuda_OpenMP-release
#######################################################
PASSED TESTS
#######################################################
cuda-9.2.88-Cuda_OpenMP-release build_time=885 run_time=384
#######################################################
FAILED TESTS
#######################################################
```

File: src/batched/KokkosBatched_Vector_SIMD.hpp [hunk 1]
```diff
@@ -135,6 +117,87 @@ namespace KokkosBatched {
...
 
+  template<>
+  class Vector<SIMD<float>,2> {
+  public:
+    using type = Vector<SIMD<float>,2>;
+    using value_type = float;
+    using mag_type = float;
+
+    enum : int { vector_length = 2 };
+    typedef float2 data_type;
```
Comment (mhoemmen) [inline src/batched/KokkosBatched_Vector_SIMD.hpp]: Could names like `float2` collide with existing types in the global namespace in e.g., CUDA?

Comment (kyungjoo-kim) [reply src/batched/KokkosBatched_Vector_SIMD.hpp]: This is intended to override the cuda builtin type (float2, doule2, float4, double4).

**Pull request merged**